src/policies.rs

```diff
@@ -1,31 +1,124 @@
 use crate::git::*;
 use crate::gpg::*;
 use crate::fs::*;
-use crate::fingerprints::*;
-
+use crate::error::CapnError;
+use crate::config::VerifyGitCommitsConfig;
 
+use git2::{Commit, Oid};
 use std::error::Error;
 use std::path::PathBuf;
 use std::time::Instant;
+use std::collections::HashSet;
 use log::*;
 
+const DONT_CARE_REF: &str = "0000000000000000000000000000000000000000";
+
 pub fn prepend_branch_name<F: Fs, G: Git>(commit_file: PathBuf) -> Result<(), Box<dyn Error>> {
-    trace!("Executing policy: prepend_branch_name");
+    debug!("Executing policy: prepend_branch_name");
     
     let git = G::new()?;
     let branch = git.current_branch()?;
     Ok(F::prepend_string_to_file(branch, commit_file)?)
 }
 
-pub fn verify_git_commits<G: Git, P: Gpg>(_new_value: &str, team_fingerprints_file: &str, keyserver: &str) -> Result<(), Box<dyn Error>> {
-    trace!("Executing policy: verify_git_commits");
+pub fn verify_git_commits<G: Git, P: Gpg>(config: &VerifyGitCommitsConfig, old_value: &str, new_value: &str,_ref_name: &str) -> Result<(), Box<dyn Error>> {
+    debug!("Executing policy: verify_git_commits");
+    
+    let git = G::new()?;
     let start = Instant::now();
+    let old_commit_id = Oid::from_str(old_value)?;
+    let new_commit_id = Oid::from_str(new_value)?;
+
+    let new_commit = git.find_commit(new_commit_id)?;
```

Comment:
Might want to put all this code which figures out which commits to check into a function to make this function a bit cleaner.